### PR TITLE
Add workflow for basic codeql analysis

### PR DESCRIPTION
**Pull request by joshuagl**
Fixes #NA

**Description of the changes being introduced by the pull request**:

Adds a workflow to leverage CodeQL analysis

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Review by jku — COMMENTED** [review]
I've looked at codeql checks at some point: very few of them seem to be relevant to our code base but I suppose there's nothing wrong with enabling this just in case?

* workflow would be more readable if generic comments and the unused abstractions  were removed (like a "strategy" and a "matrix" for our one single language, or the autobuild task for a non-compiled language... ). I'm fine with merging as is but more readable is probably better, especially considering following point
* what are the pros/cons of keeping this as separate workflow and not as just another job in ci.yml?
* what happens if the cron triggered workflow fails?

**Comment by lukpueh** [comment]
@joshuagl, do you want to finalize this PR? Should someone take over?

> I've looked at codeql checks at some point: very few of them seem to be relevant to our code base but I suppose there's nothing wrong with enabling this just in case?

@jku, do you know which checks could be relevant for our code base? Regardless, I agree that we probably can just enable it and see what it does.

> * workflow would be more readable if generic comments and the unused abstractions  were removed (like a "strategy" and a "matrix" for our one single language, or the autobuild task for a non-compiled language... ). I'm fine with merging as is but more readable is probably better, especially considering following point

I'm a bit against merging as is. We should at least 
- get the branch names right
- remove commented out code and irrelevant comments
- pin the used `codeql-action`s to sha refs

> * what are the pros/cons of keeping this as separate workflow and not as just another job in ci.yml?

I think the only reason to add the codeql job to the ci workflow would be to trigger on the same events (which would mean no cron) and execute it on the same runner (which has no clear advantage)

> * what happens if the cron triggered workflow fails?

Maybe display an alert?
https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning-alerts

**Comment by joshuagl** [comment]
Thanks for the feedback both. I'd just taken the sample workflow and opened a draft PR to see if it provided any useful results. I can clean up the PR and mark it "ready for review" as we have consensus that it's worth trying.

**Comment by joshuagl** [comment]
I _think_ this is ready for review.

AIUI the point of the scheduled analysis is so that there's an up-to-date baseline for PR's to compare against, then the analysis run on the PR can inform submitters that there are M alerts compared to N on the main branch.

**Review by jku — APPROVED** [review]
> @jku, do you know which checks could be relevant for our code base?

I don't remember details, just that most checks were relevant for e.g. web services and such. I'm sure there are things useful for us as well and there's no telling how the test set evolves in future.

Anyway, looks fine to me.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: "CodeQL Analysis"
+
+on:
+  push:
+    branches: [ develop, '*' ]
```
Comment (lukpueh) [inline .github/workflows/codeql-analysis.yml]: IIUC this means trigger on push to develop branch and all branches that don't contain a slash. Is this intended?

**Review by lukpueh — APPROVED** [review]
Thanks for cleaning this up, @joshuagl! I left one question about the trigger condition inline, but I'm fine merging as is.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (joshuagl) [reply .github/workflows/codeql-analysis.yml]: You're right (see [filter pattern cheat sheet](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet)). I had misremembered and thought we had a branch for the pre 1.0 state, but we do not. I think we only want to do CodeQL analysis on the `develop` branch.

**Comment by joshuagl** [comment]
I had to rebase on the latest develop, so squashed commits into one. The only code change since the force push was to update the branches we trigger on per https://github.com/theupdateframework/python-tuf/pull/1932#discussion_r897745187

**Pull request merged**